### PR TITLE
refactor: split remote handler + add unit tests

### PR DESCRIPTION
**Pull request by elliott-with-the-longest-name-on-github**
I'm on a slow-and-steady mission to start being able to replace some of our integration tests with unit tests. Our integration tests often contort themselves into all sorts of horrible shapes to try to assert that some internal-to-sveltekit or side-effect-y thing is happening. Part of this effort is going to be busting up some of these big "orchestrator" files into abstractions that are easier to test. The remote function server handling hadn't gotten too bad yet, but it was definitely on its way there. I split it up some:

- Split `remote.js` into more-focused `remote/` modules. I think most of the code is clearer now.
- Added shared server mock factories in `test/mocks/server.js` -- we had a couple of places where we were partially mock-and-cast-ing our request state stuff, this just kinda standardizes it
- Unit tests for the bits that previously needed a browser: keep-alive timing (fake timers), dispatch per type, the collect_remote_data race/redirect/private-fn edge cases, keyed form.for() (incl. slash keys), `requested()` limits, and OTEL span emission
- Tagged some unit tests with `UNIT-MIGRATION-CANDIDATE` for later; when I've had time to do this to more of the codebase and we're still stable for a while, we can sweep through and remove them

### CODE REVIEW

File: packages/kit/src/exports/hooks/sequence.spec.js [hunk 1]
```diff
@@ -1,29 +1,17 @@
...
-
-const dummy_event = vi.hoisted(
-	() =>
-		/** @type {RequestEvent} */ ({
-			tracing: {
-				root: {}
-			}
-		})
-);
+import { create_mock_event } from '../../../test/mocks/server.js';
```
Comment (Rich-Harris) [inline packages/kit/src/exports/hooks/sequence.spec.js:3]: can we create an internal import for these like `#test/mocks/server`?

File: packages/kit/src/exports/hooks/sequence.spec.js [hunk 2]
```diff
@@ -1,29 +1,17 @@
...
+
+const dummy_event = create_mock_event();
 
 vi.mock(import('@sveltejs/kit/internal/server'), async (actualPromise) => {
 	const actual = await actualPromise();
+	const { create_mock_event, create_mock_state } = await import('../../../test/mocks/server.js');
```
Comment (Rich-Harris) [inline packages/kit/src/exports/hooks/sequence.spec.js:9]: why do we need a static import _and_ a dynamic one?

File: packages/kit/src/runtime/server/remote/call.spec.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	create_mock_remote,
+	create_mock_request,
+	create_mock_state
+} from '../../../../test/mocks/server.js';
+
+describe('handle_remote_call', () => {
+	beforeAll(() => {
+		// referenced as a bare global by `handle_error_and_jsonify`
+		// @ts-expect-error
+		globalThis.__SVELTEKIT_DEV__ = false;
```
Comment (Rich-Harris) [inline packages/kit/src/runtime/server/remote/call.spec.js:19]: depending on how often we find ourselves doing this it might be worth creating a helper for it

File: packages/kit/src/runtime/server/remote/call.spec.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+import { BINARY_FORM_CONTENT_TYPE, serialize_binary_form } from '../../form-utils.js';
+import {
+	create_mock_event,
+	create_mock_internals,
+	create_mock_manifest,
+	create_mock_options,
+	create_mock_remote,
+	create_mock_request,
+	create_mock_state
+} from '../../../../test/mocks/server.js';
```
Comment (Rich-Harris) [inline packages/kit/src/runtime/server/remote/call.spec.js:13]: might feel nicer like this:

```suggestion
import * as mocks from '#test/mocks/server';
```

followed by

```js
const internals = mocks.create_internals(...);
```

File: packages/kit/src/runtime/server/remote/call.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
+/** @import { RequestEvent, SSRManifest } from '@sveltejs/kit' */
+/** @import { RemoteFunctionData, RemoteInternals, RequestState, SSROptions } from 'types' */
+
+import { error } from '@sveltejs/kit';
+import { Redirect } from '@sveltejs/kit/internal';
+import { with_request_store, merge_tracing } from '@sveltejs/kit/internal/server';
+import { record_span } from '../../telemetry/record_span.js';
+import { handle_error_and_jsonify } from '../utils.js';
+import { collect_remote_data } from './collect.js';
+import { run_command, run_form, run_prerender, run_query, run_query_batch } from './dispatch.js';
```
Comment (Rich-Harris) [inline packages/kit/src/runtime/server/remote/call.js:10]: what's the argument for separating `call.js` and `dispatch.js`? wouldn't it be easier to put the contents of `dispatch.js` in here?

**Comment by Rich-Harris** [comment]
this is probably going to cause some merge conflict misery — it might be better to wait until more of the PR queue gets merged before going much further. any PRs targeting `main` that touch remote functions are going to get annoying to untangle

**Comment by Rich-Harris** [comment]
Overall I'm happy that we're moving in the direction of faster tests, but after a quick first pass through I worry about a couple of things (besides the aforementioned merge conflicts):

- we're testing a lot of internal implementation details that _don't matter_. as those implementation details evolve, we will need to update all our tests (or worse, we will be discouraged from improving the implementation because of the resulting headache)
- some of the tests are truly just noise — we absolutely don't need unit tests for things like `assert_form_content_type` or `get_payload` for example
- we're using mocks a _lot_, and mocks have always felt like a bit of a code smell to me. unavoidable at times but not a reliable substitute for reality

Obviously what we have now is a PITA and I'm not defending it. But I think the sweet spot is much more coarse-grained than this — for example it would be nice to test something like 'given an app with the following manifest, assert that if I call `server.respond(...)` then the following query results are inlined into the response' without involving Playwright. Perhaps replacing the existing `server.test.js` files would be an ideal place to start
